k8s/scripts/sysbox-deploy-k8s.sh

```diff
@@ -948,6 +948,28 @@ function do_distro_adjustments() {
 	sed -i '/^kernel.unprivileged_userns_clone/ s/^#*/# /' ${sysbox_artifacts}/systemd/99-sysbox-sysctl.conf
 }
 
+# determines if running on a GKE cluster by checking metadata endpoint
+function check_gke() {
+	is_cluster=$(curl -s -o /dev/null \
+	 		-w "%{http_code}" \
+	   		--connect-timeout 1 \
+	     		-H "Metadata-Flavor: Google" \
+	       		169.254.169.254/computeMetadata/v1/instance/attributes/cluster-name)
+	if [ $is_cluster -ne 200 ]; then
```

Comment:
If possible, please add a comment here on why we are looking for return code 200 (similar to what you wrote down in this PR's conversation thread). Thanks!

Reply:
Good call out @ctalledo, I added a comment explaining the logic